### PR TITLE
Exported types for the permission router metadata endpoint

### PR DESCRIPTION
**Pull request by HHogg**
Signed-off-by: Harry Hogg <hhogg@spotify.com>

## Hey, I just made a Pull Request!

Exported some types for the response of the .metadata endpoint of the permission router. 

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/permission-node/api-report.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ import { DefinitivePolicyDecision } from '@backstage/plugin-permission-common';
 import { EvaluatorRequestOptions } from '@backstage/plugin-permission-common';
 import express from 'express';
 import { IdentifiedPermissionMessage } from '@backstage/plugin-permission-common';
+import { JsonSchema7Type } from 'zod-to-json-schema/src/parseDef';
```
Comment (HHogg) [inline plugins/permission-node/api-report.md:17]: Not sure about this import tbh. An alternative might be `ReturnType<typeof zodToJsonSchema>`

**Review by Joonpark13 — APPROVED** [review]

File: plugins/permission-node/api-report.md (hunk 1 above)
Comment (Rugvip) [reply plugins/permission-node/api-report.md:17]: Think this is prolly better. Is the schema type omportant at all though? Could it be an opaque `object`?

File: plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts [hunk 2]
```diff
@@ -102,6 +103,29 @@ export type ApplyConditionsResponse = {
...
+  resourceType: string;
+  paramsSchema?: JsonSchema7Type;
+};
+
+/**
+ * Response type for the .metadata endpoint.
+ *
+ * @public
+ */
+export type MetaDataResponse = {
```
Comment (Rugvip) [inline plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts:124]: Should this be `PermissionsMetadataResponse` instead?

**Review by Rugvip — COMMENTED** [review]
👍, two small thoughts

File: plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts (hunk 2 above)
Comment (HHogg) [reply plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts:124]: I had an internal battle with this one, I agree it should be prefixed however there's another exported type in this file called `ApplyConditionsResponse` so I went with consistency. What do you think? Go for the prefix or match it up with the other one.

**Review by vinzscam — APPROVED** [review]

File: plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts (hunk 2 above)
Comment (Rugvip) [reply plugins/permission-node/src/integration/createPermissionIntegrationRouter.ts:123]: Alright, let's keep it short. Could you switch it to `MetadataResponse` though? That's the casing we use everywhere else.

**Review by Rugvip — CHANGES_REQUESTED** [review]

**Review by Rugvip — APPROVED** [review]
Alright, let's :shipit:!

**Pull request merged**